### PR TITLE
Dashboard: Fix Editing a panel with auto-refresh enabled auto-refresh in the edit screen 

### LINKED ISSUE #21764: Dashboard: Editing a panel with auto-refresh enabled auto-refresh in the edit screen
**What happened**:
If a dashboard has auto-refresh on it, and you edit a panel, then the panel edit screen also has has auto-refresh set.
If one edits the query then, and the panel refreshes while the syntax is incorrect, then when it auto-refreshes it puts an error marker in the top left of the graph, but also moves the cursor to the start of the query again.

**What you expected to happen**:
The edit screen for the panel should not have auto-refresh enabled by default. 

**How to reproduce it (as minimally and precisely as possible)**:
visit https://play.grafana.org/d/000000029/prometheus-demo-dashboard?orgId=1&refresh=30s
edit "network usage" panel
Change irate for random text. 
wait 30 seconds. 

**Anything else we need to know?**:
According to the customer, this started at version 6.3.0 

**Environment**:
- Grafana version:
>=6.3.0

### PR DESCRIPTION
**Pull request by axelavargas**
**What this PR does / why we need it**:

When a Dashboard has the auto-refresh option set, then the panel edit screen also had it, it was causing unexpected refresh for the users if, for example, they were setting a query. 

**Approach**

The approach in this PR:
- **Hide auto-refresh option(The interval one) on panel edit**, the reason behind hiding the option instead of not having the auto-refresh enabled by default is to avoid keeping track of the auto-refresh values changed per panel. (currently, we only support global auto-refresh that apply to all the panels)

Before:

Screenshot from 2021-05-17 15-03-51

**This is how will look like with the option hidden, the refresh (sync button) is there:**

Screenshot from 2021-05-17 15-00-51

**Which issue(s) this PR fixes**:

Fixes #21764

**Special notes for your reviewer**:

I am lacking some context in the project, so it might be worth thinking on missing edge cases where this solution does not work

### CODE REVIEW

File: public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx [hunk 1]
```diff
@@ -88,6 +88,7 @@ export class DashNavTimeControls extends Component<Props> {
     const timePickerValue = getTimeSrv().timeRange();
     const timeZone = dashboard.getTimezone();
     const styles = getStyles();
+    const hideRefreshPicker = !dashboard.panelInEdit?.isEditing;
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx:91]: Hmm, can we reverse the naming of this variable? Shouldn't it be showRefreshPicker?

File: public/app/features/dashboard/state/DashboardModel.test.ts [hunk 2]
```diff
@@ -849,8 +859,15 @@ describe('exitPanelEditor', () => {
...
+    it('then refresh property is set to previousAutoRefresh property', () => {
+      const { dashboard } = getTestContext(true);
+      dashboard.exitPanelEditor();
+
+      expect(timeSrvMock.setAutoRefresh).toHaveBeenCalledWith('5s');
+    });
+
     describe('and there is a change that affects all panels', () => {
-      it('then startRefresh is not called', () => {
+      it('then startRefresh is called', () => {
```
Comment (hugohaggmark) [inline public/app/features/dashboard/state/DashboardModel.test.ts:870]: Nice catch!

File: public/app/features/dashboard/state/DashboardModel.test.ts [hunk 3]
```diff
@@ -7,8 +7,16 @@ import { createAdHocVariableAdapter } from '../../variables/adhoc/adapter';
...
+import { setTimeSrv, TimeSrv } from '../services/TimeSrv';
 
 jest.mock('app/core/services/context_srv', () => ({}));
+
+const timeSrvMock = ({
+  setAutoRefresh: jest.fn(),
+} as unknown) as TimeSrv;
+
+setTimeSrv(timeSrvMock);
+
```
Comment (hugohaggmark) [inline public/app/features/dashboard/state/DashboardModel.test.ts:19]: nit: maybe the `setTimeSrv(timeSrvMock);` should be set within the `getTestContext` functions so it's not set in the global scope?

File: public/app/features/dashboard/state/DashboardModel.ts [hunk 4]
```diff
@@ -356,6 +359,8 @@ export class DashboardModel {
   }
 
   initEditPanel(sourcePanel: PanelModel): PanelModel {
+    this.previousAutoRefresh = this.refresh;
+    getTimeSrv().setAutoRefresh('');
```
Comment (hugohaggmark) [inline public/app/features/dashboard/state/DashboardModel.ts:363]: nit: maybe this logic belongs to `TimeSrv`,  we could have an `pauseAutoRefresh` and `resumeAutoRefresh` function that handles this logic. But at the same time it feels strange that TimeSrv should mutate dashboard state directly.

File: public/app/features/dashboard/components/DashNav/DashNavTimeControls.test.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  it('should not render RefreshPicker in panel edit', () => {
+    const panel: any = { destroy: jest.fn(), isEditing: true };
+    dashboardModel.startRefresh = jest.fn();
+    dashboardModel.panelInEdit = panel;
+    const container = render(
+      <DashNavTimeControls dashboard={dashboardModel} onChangeTimeZone={jest.fn()} key="time-controls" />
+    );
+    expect(container.queryByLabelText(/RefreshPicker run button/i)).not.toBeInTheDocument();
+  });
+});
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/DashNav/DashNavTimeControls.test.tsx:49]: Skilled work adding these tests 🎉

**Review by hugohaggmark — APPROVED** [review]
Skilled work on this and works as expected. Outstanding that you added tests, I left some small comments for consideration.

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 4 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts:363]: > nit: maybe this logic belongs to `TimeSrv`, we could have an `pauseAutoRefresh` and `resumeAutoRefresh` function that handles this logic. But at the same time it feels strange that TimeSrv should mutate dashboard state directly. (But it already does)

**Review by torkelo — COMMENTED** [review]
Great job! beyond the small comments  Hugo added, this looks great!

File: public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx (hunk 1 above)
Comment (axelavargas) [reply public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx]: Thanks for the suggestion :D, I also did a small fix here as well,  this component also had the `refresh/sync` button that needs to be always present https://github.com/grafana/grafana/pull/34128/commits/02f4f35075001434564b9a2244a10706936f7b8c#diff-aac52f3778332268268f492b861c176a728fa4e7f05a4a5574228ec00920482aR91

File: public/app/features/dashboard/state/DashboardModel.test.ts (hunk 3 above)
Comment (axelavargas) [reply public/app/features/dashboard/state/DashboardModel.test.ts]: indeed, a good idea 🙌🏾 https://github.com/grafana/grafana/pull/34128/commits/02f4f35075001434564b9a2244a10706936f7b8c#diff-0338d437d3a32fb5066ab71687aecdaf0229a778cde30e595af9ff1a39f152d1R823 🚀

File: public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx]: Looks great 👍

File: public/app/features/dashboard/state/DashboardModel.ts (hunk 4 above)
Comment (axelavargas) [reply public/app/features/dashboard/state/DashboardModel.ts]: I agree with you,  this logic belongs to the `TimeSrv`, I  did a small refactor in the code https://github.com/grafana/grafana/pull/34128/commits/e553524da6f8cbaea5c20c12e8f270ccbade62b8 and now we are controlling the value of the `previousAutoRefresh` in the `TimeSrv`

Comment (hugohaggmark) [reply public/app/features/dashboard/state/DashboardModel.ts]: Beautiful 🎉

**Comment by hugohaggmark** [comment]
@axelavargas feel free to `Squash and merge` when you're finished with this PR 👍

**Pull request merged**